### PR TITLE
Dmitri/forwarder config

### PR DESCRIPTION
**Pull request by a-palchikov**
This PR adds support for log forwarder configuration to the `log-collector`. In order to achieve this, `log-tailer` has moved inside the `log-collector` container to be able to directly control `rsyslogd`.

The new `/forwarders` endpoint accepts PUT requests with a list of forwarders which it then writes as `rsyslogd` .conf files and restarts the service.

### CODE REVIEW

File: cmd/wstail/forwarders.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			return trace.Wrap(err)
+		}
+		if path == rsyslogConfigDir {
+			return nil
+		}
+		if info.IsDir() {
+			return filepath.SkipDir
+		}
+		log.Infof("removing %v", path)
+		return os.Remove(path)
```
Comment (klizhentas) [inline cmd/wstail/forwarders.go]: `return trace.Wrap(os.Remove(path), "failed removing %v", path`

File: cmd/wstail/forwarders.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	// Write new forwarder configuration
+	for _, forwarder := range forwarders {
+		f, err := os.Create(path(forwarder))
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		_, err = f.WriteString(config(forwarder))
+		f.Close()
```
Comment (klizhentas) [inline cmd/wstail/forwarders.go]: check for error

File: cmd/wstail/forwarders.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		log.Warningf("failed to delete forwarder configuration files: %v", err)
+	}
+
+	// Write new forwarder configuration
+	for _, forwarder := range forwarders {
+		f, err := os.Create(path(forwarder))
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		_, err = f.WriteString(config(forwarder))
```
Comment (klizhentas) [inline cmd/wstail/forwarders.go:55]: check for error

File: cmd/wstail/forwarders.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		f.Close()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	// Reload rsyslogd
+	if out, err := exec.Command("/etc/init.d/rsyslog", "restart").CombinedOutput(); err != nil {
```
Comment (klizhentas) [inline cmd/wstail/forwarders.go]: what's the base container for this? are we using @alexey-medvedchikov 's containers?

Comment (klizhentas) [inline cmd/wstail/forwarders.go]: why is this not a constant, while `const rsyslogConfigDir` is?

Comment (a-palchikov) [inline cmd/wstail/forwarders.go]: the base container is `gravitational/debian-grande`

Comment (klizhentas) [inline cmd/wstail/forwarders.go]: so you are using `dumb-init` there or something else?

Comment (klizhentas) [inline cmd/wstail/forwarders.go]: how is reload is going to work if you don't use init.d and using dumb-init?

File: cmd/wstail/forwarders.go (hunk 3 above)
Comment (a-palchikov) [inline cmd/wstail/forwarders.go:55]: checking down the line

File: cmd/wstail/forwarders.go (hunk 4 above)
Comment (sofuture) [inline cmd/wstail/forwarders.go]: I suspect the files are there, but yeah, won't work. Might be able to just `hup` the rsyslog pid though.

**Comment by klizhentas** [comment]
Also, I think you should make API versioned

**Pull request merged**